app.py

```diff
@@ -197,113 +195,115 @@ def insert_titles_by_rating_records(session, preparedInsertRating):
-            session, TITLE_PULP_FICTION, prepared_read_director_by_title_primary))
-
-        print ""
-        print "Updating record with read - step 6"
-        update_director_in_primary_by_title(session, SHOW_ID_PULP_FICTION,
-                                            TITLE_PULP_FICTION,
-                                            ['Quentin Jerome Tarantino'], prepared_update_director)
-
-        print ""
-        print_all(read_director_from_primary_by_title(
-            session, TITLE_PULP_FICTION, prepared_read_director_by_title_primary))
-
+        try:
+            print ""
+            print "Creating keyspace - step 2"
+
+            create_keyspace(session)
+
+            print ""
+            print "Creating tables - step 3"
+
+            create_primary(session)
+            create_titles_by_date(session)
+            create_titles_by_rating(session)
+
+            print ""
+            print "Creating prepared statements - step 4"
+            prepared_insert_primary = prepare_inserts_primary(session)
+            prepared_insert_date = prepare_inserts_date(session)
+            prepared_insert_rating = prepare_inserts_rating(session)
+            prepared_read_director_by_title_primary = prepare_reads_director_by_title_primary(
+                session)
+            prepared_read_all_by_title_primary = prepare_reads_all_by_title_primary(
+                session)
+            prepared_update_director = prepare_update_director(session)
+
+            print ""
+            print "Inserting records - step 5"
+            insert_primary_records(session, prepared_insert_primary)
+            insert_titles_by_date_records(session, prepared_insert_date)
+            insert_titles_by_rating_records(session, prepared_insert_rating)
+
+            print ""
+            print "Reading records - step 6"
+            print_all(read_all(session, TABLE_NETFLIX_PRIMARY))
+            print ""
+            print_all(read_all(session, TABLE_NETFLIX_TITLES_BY_RATING))
+            print ""
+            print_all(read_all(session, TABLE_NETFLIX_TITLES_BY_DATE))
+            print ""
+            print_all(read_all_primary_by_title(
+                session, TITLE_PULP_FICTION, prepared_read_all_by_title_primary))
+            print ""
+            print_all(read_director_from_primary_by_title(
+                session, TITLE_PULP_FICTION, prepared_read_director_by_title_primary))
+
+            print ""
+            print "Updating record with read - step 7"
+            update_director_in_primary_by_title(session, SHOW_ID_PULP_FICTION,
+                                                TITLE_PULP_FICTION,
+                                                ['Quentin Jerome Tarantino'], prepared_update_director)
+
+            print ""
+            print_all(read_director_from_primary_by_title(
+                session, TITLE_PULP_FICTION, prepared_read_director_by_title_primary))
+
+        finally:
```

Comment:
It may be cleaner to have just one try/except/finally instead of the inner try/finally.

Reply:
It would appear cleaner, yes -- but having just one gives a warning in the finally block that `cluster` may not be defined. The inner block is a try/finally starts after we're connected to the cluster and exists only to guarantee that we disconnect. The outer block is a try/except and exists only to capture errors.